packages/testcontainers/src/containers/NativeChainContainer.ts

```diff
@@ -0,0 +1,238 @@
+      '-fortcanninghillheight=10',
+      '-fortcanningroadheight=11',
+      '-fortcanningcrunchheight=12',
+      '-fortcanningspringheight=13',
+      '-fortcanninggreatworldheight=14',
+      '-fortcanningepilogueheight=15',
+      '-grandcentralheight=16',
+      '-regtest-skip-loan-collateral-validation',
+      '-regtest-minttoken-simulate-mainnet=0'
+    ],
+    mainnet: [],
+    devnet: [],
+    masternode: [
+      '-dummypos=0',
+      '-spv=1',
+      '-anchorquorum=2'
+    ]
+  }
+
+  /**
+   * Convenience Cmd builder
+   */
+  protected generateCmd (): string[] {
+    const { defaultCmd } = NativeChainContainer
+
+    const authCmd = [
+      `-rpcuser=${this.rpcUser}`,
+      `-rpcpassword=${this.rpcPassword}`
+    ]
+
+    const masterNodeOperator = (this.masterNodeKey != null)
+      ? [`-masternode_operator=${this.masterNodeKey.operator.address}`]
+      : []
+
+    return [
+      ...defaultCmd.prepend,
+      ...authCmd,
+      ...defaultCmd[this.blockchainNetwork.name],
+      ...this.hasPreconfiguredMasternode ? defaultCmd.masternode : [],
+      ...masterNodeOperator,
+      ...this.addedCmds
+    ]
+  }
+
+  private readonly addedCmds: string[] = []
+  public addCmd (newCmd: string): this {
+    this.addedCmds.push(newCmd)
+    return this
+  }
+
+  /**
+   * Create container and start it immediately waiting for NativeChain to be ready
+   */
+  public async start (): Promise<StartedNativeChainContainer> {
+    this.withExposedPorts(...(this.hasExposedPorts ? this.ports : Object.values(this.blockchainNetwork.ports)))
+      .withName(this.name ?? this.generateName())
+      .withCommand(this.command.length > 0 ? this.command.concat(this.addedCmds) : this.generateCmd())
+
+    const {
+      rpcUser,
+      rpcPassword,
+      blockchainNetwork,
+      masterNodeKey
+    } = this
+
+    return await new StartedNativeChainContainer(
+      await super.start(),
+      {
+        rpcUser,
+        rpcPassword,
+        blockchainNetwork,
+        masterNodeKey
+      }
+    ).withPrivateKeys()
+  }
+}
+
+export interface StartedContainerConfig {
+  rpcUser: string
+  rpcPassword: string
+  blockchainNetwork: BlockchainNetwork
+  masterNodeKey?: MasterNodeKey
+}
+
+export class StartedNativeChainContainer extends AbstractStartedContainer {
+  rpc = new NativeChainRpc(this)
+  waitFor = new NativeChainWaitFor(this)
+
+  constructor (
+    startedTestContainer: StartedTestContainer,
+    private readonly config: StartedContainerConfig
+  ) {
+    super(startedTestContainer)
+  }
+
+  get rpcUser (): string {
+    return this.config.rpcUser
+  }
+
+  get rpcPassword (): string {
+    return this.config.rpcPassword
+  }
+
+  get blockchainNetwork (): BlockchainNetwork {
+    return this.config.blockchainNetwork
+  }
+
+  get masterNodeKey (): MasterNodeKey | undefined {
+    return this.config.masterNodeKey
+  }
+
+  public async withPrivateKeys (): Promise<this> {
+    if (this.masterNodeKey != null) {
```

Comment:
```suggestion
  public async withMasternode (keys: MasterNodeKey): Promise<this> {
```

How about something like this?